rfd/0084-oracle-db-access-integration.md

```diff
@@ -0,0 +1,105 @@
+---
+authors: Marek Smoliński (marek@goteleport.com)
+state: draft
+---
+
+# RFD 84 - Teleport Oracle Access Integration
+
+## What
+
+
+This RFD outlines the scope of Oracle integration with Teleport Database Access.
+Oracle DB Access differs from currently supported database protocols by Teleport Database Access.
+Namely, Oracle DB protocol is private and there is not any official documentation describing the Oracle protocol.
+
+## Why
+
+Adding Oracle support to Teleport Database Access would allow our shared customer bases to take advantage of Teleport's access controls and audit logging capabilities when accessing their Oracle databases.
+
+* Administrators will be able to control access to their entire fleet of database servers through their identity provider with SSO.
+* Users will be able to connect to Oracle databases with the tools they're familiar with without having to deal with passwords or shared secrets.
+* Auditors will be able to view the database activity and tie it to a particular user identity within an organization.
+
+
+# Scope of Integration
+-- **Teleport as Oracle Access Proxy**: Teleport should be able to act like a proxy between the incoming Oracle client connection and connection to Oracle Server where the Teleport will terminate the incoming TLS connection and establish a new TLS connection to the Oracle Server using a new TLS Certificate and forward the traffic between Oracle client and server.
+- **Audit Logging**: Teleport needs to be able to inspect the Oracle wire protocol to provide Teleport audit logs and audit client interaction with Oracle database.
+
+
+### TLS Termination of Incoming connection:
+Teleport needs to be able to TLS-terminate incoming Oracle client connection and reestablish a new TLS connection that uses Teleport-signed client certificate to the Oracle Server. Oracle database server will need to be configured to trust Teleport's certificate authority to be able to validate client connections.
+
+
+## Details
+
+### Teleport Database Access Configuration:
+
+#### Oracle Client:
+
+Oracle clients support TLS connections to the Oracle Server by using a custom container called [Oracle Wallet](https://docs.oracle.com/cd/E92519_02/pt856pbr3/eng/pt/tsvt/concept_UnderstandingOracleWallet.html#:~:text=Oracle%20Wallet%20is%20a%20container,is%20used%20for%20security%20credentials.
```

Comment:
Do we have a specific set of Oracle versions we will support, ex: 15+?

Reply:
Yes, I'm in the process of testing the integration with other Oracle versions. I will updated the RFD soon.

Reply:
@stevenGravy 
I have added a section about Oracle server requirement: https://github.com/gravitational/teleport/pull/16435/files#diff-a348bce1705dbd09c2ec0ab96d7e4fa49b7f6f5b4413cd9e07fe0986e1bd904dR40

Reply:
thanks!